src/main/java/io/radanalytics/operator/app/AppOperator.java

```diff
@@ -44,12 +61,28 @@ protected void onAdd(SparkApplication app) {
         KubernetesResourceList list = deployer.getResourceList(app, namespace);
         client.resourceList(list).inNamespace(namespace).createOrReplace();
         updateStatus(app, "ready" );
+        put(app);
+    }
+
+    @Override
+    protected void onModify(SparkApplication newApp) {
+
+        // This comparison works to rule out a change in status because
+        // we added the status block in the AbstractOperator universally,
+        // ie it is not actually included in the SparkApplication type
+        // definition generated from json. If that ever changes, then
+        // this comparison will have to be a little smarter.
```

Comment:
maybe throw a `// TODO fix this if it changes` just so this doesn't get lost in the shuffle?

Reply:
++ good idea